server/scripts/import-tests/index.js

```diff
@@ -159,277 +151,206 @@ const ariaat = {
-                        }
-                    }
-                );
-            } else {
-                console.error(
-                    `WARNING: The design pattern link is not valid for ${exampleName}. Not writing to database.`
-                );
+        const getReferenceUrl = referenceLine => {
+            let url = null;
+            if (referenceLine.length) {
+                const [referenceType, link] = referenceLine[0].split(',');
+                if (validUrl.isUri(link)) url = link;
+                else
+                    console.error(
+                        `WARNING: The ${referenceType} link ${link} is not valid for ${exampleName}. Not writing to database.`
+                    );
             }
-        }
+            return url;
+        };
+
+        const exampleUrl = getReferenceUrl(exampleRefLine);
+        const designPattern = getReferenceUrl(practiceGuidelinesRefLine);
+
+        let parsedTest = {
+            title: '',
+            gitRepo: ariaAtRepo,
+            directory: exampleDir,
+            minimumInputCount: 0,
+            maximumInputCount: 0,
+            testActions: [],
+            designPattern
+        };
+
+        // checking to see if unique testPlan row
+        const testPlanResult = await client.query(
+            'SELECT id, revision FROM "TestPlan" WHERE revision=$1 and "exampleUrl"=$2',
+            [commitHash, exampleUrl]
+        );
 
-        return example.id;
+        const testPlan = testPlanResult.rowCount
+            ? testPlanResult.rows[0]
+            : await db.TestPlan.create({
+                  title: exampleName,
+                  publishStatus: 'draft',
+                  revision: commitHash,
+                  sourceGitCommit: commitMsg,
+                  exampleUrl,
+                  createdAt: commitDate,
+                  parsedTest
+              });
+        return testPlan.id;
     },
 
     /**
-     * Gets the test id and adds the test record if it does not exist
-     * @param {string} testFullName - the name of the test
-     * @param {string} file - the relative path to the test in the repo
-     * @param {int} exampleID - foreign key into the apg_example table
-     * @param {int} testVersionID - foreign key into the test_version table
-     * @param {int} executionOrder - order of the tests (within APG pattern)
-     * @return {int} id
+     * Checks TestPlan.parsedTest.testActions to see if it has the relevant test actions to run the test and inserts it if not
+     * @param {string} testName - the name of the test
+     * @param {string} file - the relative path to the test file in the repository (ideally {@link https://github.com/w3c/aria-at.git})
+     * @param {number} testPlanId - TestPlan.id to be queried to update the TestPlan.parsedTest.testActions if necessary
+     * @param {string} commitHash - the hash of the latest version of tests pulled from the repository (ideally {@link https://github.com/w3c/aria-at.git})
```

Comment:
I think we should make sure this matches the TestPlan model, so `sourceGitCommit`.

Given the fact we need the message too, it could be `sourceGitCommitHash`, in which case I'd definitely prefer to update the model from `sourceGitCommit` to `sourceGitCommitHash` as well.

Reply:
I was puzzled at first because `revision` is being used as the hash here and `sourceGitCommit` as the message. But I follow now. I agree.

1. Re-purposing `revision` to be the latest tag found for the commit being pulled (may need to look specifically for semver-ed patterns)
2. `sourceGitCommit` -> `sourceGitCommitHash`
3. Adding `sourceGitCommitMessage` to the TestPlan schema

Reply:
Awesome! Thanks for adding the message to the TestPlan too, that's a great solution.

As for revision, I think the hope is for it to come from a future version of the ARIA-AT repo, but it currently doesn't exist, haha. Maybe it will be stored with the test source as metadata on the test plan or something. Anyway, maybe the right move is to remove it until it's actually available.

Reply:
Yes, right now we only have the git commit hash and message to track changes.  if we include `revision` (for future-proofing) I think it should just duplicate the commit hash (since git tags are not really used by the aria-at repo). or, we can remove `revision` and add with a migration later.